### PR TITLE
UIQM-6 Delete a field/row in quickMARC

### PR DESCRIPTION
**Pull request by uladislausamets**
## Purpose
https://issues.folio.org/browse/UIQM-6

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: translations/ui-quick-marc/en.json [hunk 1]
```diff
@@ -45,5 +45,7 @@
   "record.fixedField.Proj": "Proj",
 
   "record.error.title.multiple": "Record cannot be saved with more than one field 245.",
-  "record.error.title.empty": "Record cannot be saved without field 245."
+  "record.error.title.empty": "Record cannot be saved without field 245.",
+  "record.delete.confirmLabel": "Yes",
```
Comment (NikitaSedyx) [inline translations/ui-quick-marc/en.json:49]: please add a line

File: src/QuickMarcEditor/QuickMarcEditorRows/utils.js [hunk 2]
```diff
@@ -17,3 +17,7 @@ export const hasIndicatorException = recordRow => INDICATOR_EXEPTION_ROWS.includ
 const ADD_EXCEPTION_ROWS = [LEADER_TAG, '001', '005', '008'];
 
 export const hasAddException = recordRow => ADD_EXCEPTION_ROWS.includes(recordRow.tag);
+
+const DELETE_EXCEPTION_ROWS = [LEADER_TAG, '001', '005', '008', '999'];
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/utils.js:21]: ticket says about `999ff`, please look at isReadOnly - seems you'll need to add common function

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 3]
```diff
@@ -89,19 +110,40 @@ const QuickMarcEditorRows = ({ name, fields, mutators: { addRecord } }) => {
...
                       data-index={idx}
                       icon="plus-sign"
                       onClick={addNewRow}
                     />
+                  )
+                }
+                {
+                  !withDeleteRowAction && (
+                    <IconButton
+                      data-test-add-row
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:125]: remove

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 4]
```diff
@@ -1,32 +1,53 @@
...
     addRecord({ index: +target.dataset.index });
   }, [addRecord]);
 
+  const showDeleteConfirmation = useCallback(({ target }) => {
+    setRemoveIndex(+target.dataset.index);
+    toggleRemoveModal();
+  }, [setRemoveIndex]);
+
+  const confirmDeletion = useCallback(() => {
+    console.log('removeIndex', removeIndex);
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:38]: !

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 5]
```diff
@@ -1,32 +1,52 @@
...
 
+  const showDeleteConfirmation = useCallback(({ target }) => {
+    setRemoveIndex(+target.dataset.index);
+    toggleRemoveModal();
+  }, [setRemoveIndex]);
+
+  const confirmDeletion = useCallback(() => {
+    deleteRecord({ index: removeIndex });
+    toggleRemoveModal();
+  }, [deleteRecord, toggleRemoveModal]);
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:40]: seems it should depend on `removeIndex`

File: src/QuickMarcEditor/QuickMarcEditorRows/utils.js [hunk 6]
```diff
@@ -14,6 +14,16 @@ const INDICATOR_EXEPTION_ROWS = [LEADER_TAG, '001', '002', '003', '004', '005',
...
-const ADD_EXCEPTION_ROWS = [LEADER_TAG, '001', '005', '008'];
+const ADD_AND_DELETE_EXCEPTION_ROWS = [LEADER_TAG, '001', '005', '008'];
 
-export const hasAddException = recordRow => ADD_EXCEPTION_ROWS.includes(recordRow.tag);
+export const hasAddException = recordRow => ADD_AND_DELETE_EXCEPTION_ROWS.includes(recordRow.tag);
+
+export const hasDeleteException = recordRow => (
+  ADD_AND_DELETE_EXCEPTION_ROWS.includes(recordRow.tag)
+  || (
+    recordRow.tag === '999'
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/utils.js:24]: it's copy paste....
you again skip comments

Comment (uladislausamets) [reply src/QuickMarcEditor/QuickMarcEditorRows/utils.js:24]: It was misunderstanding from my side, do you mean that I need to separate part that checks 999ff row?

Comment (NikitaSedyx) [reply src/QuickMarcEditor/QuickMarcEditorRows/utils.js:24]: yes

File: src/QuickMarcEditor/QuickMarcEditorRows/utils.js [hunk 7]
```diff
@@ -1,19 +1,24 @@
...
+import { is999ffRow } from '../../common/utils';
 
 export const isReadOnly = recordRow => (
   recordRow.tag === '001'
   || (
-    recordRow.tag === '999'
-    && recordRow.indicators
-    && recordRow.indicators[0] === 'f'
-    && recordRow.indicators[0] === 'f'
+    is999ffRow(recordRow)
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/utils.js:7]: please rename to `isLastRecord`

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 8]
```diff
@@ -1,32 +1,52 @@
...
 
+  const showDeleteConfirmation = useCallback(({ target }) => {
+    setRemoveIndex(+target.dataset.index);
+    toggleRemoveModal();
+  }, [setRemoveIndex]);
+
+  const confirmDeletion = useCallback(() => {
+    deleteRecord({ index: removeIndex });
+    toggleRemoveModal();
+  }, [removeIndex]);
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:40]: what about the rest depths?

File: src/common/utils/isLastRecord.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+export const isLastRecord = recordRow => {
```
Comment (NikitaSedyx) [inline src/common/utils/isLastRecord.js:1]: why is it moved to common folder? it's used only for marc rows

Comment (uladislausamets) [reply src/common/utils/isLastRecord.js]: It moved to utils as a common function

File: src/QuickMarcEditor/QuickMarcEditorRows/utils.js [hunk 10]
```diff
@@ -1,19 +1,34 @@
...
+  return (
     recordRow.tag === '999'
     && recordRow.indicators
     && recordRow.indicators[0] === 'f'
     && recordRow.indicators[0] === 'f'
+  );
+};
+
+export const isReadOnly = recordRow => (
+  recordRow.tag === '001'
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/utils.js:13]: ` recordRow.tag === '001' || isLastRecord(recordRow)`

File: test/bigtest/tests/QuickMarcEditor-test.js [hunk 11]
```diff
@@ -63,4 +63,21 @@ describe('Quick MARC editor', () => {
       expect(quickMarcEditor.editorRows().length).to.be.equal(records.length + 2);
     });
   });
+  describe('open remove record confirmation modal', () => {
```
Comment (NikitaSedyx) [inline test/bigtest/tests/QuickMarcEditor-test.js:66]: please separate test blocks by empty line

Comment (NikitaSedyx) [reply test/bigtest/tests/QuickMarcEditor-test.js:66]: it should be smth like 'when remove button is pressed'

File: test/bigtest/tests/QuickMarcEditor-test.js [hunk 12]
```diff
@@ -63,4 +63,25 @@ describe('Quick MARC editor', () => {
...
+    beforeEach(async function () {
+      await quickMarcEditor.removeButton.click();
+    });
+
+    it('should be render confirmation modal', () => {
+      expect(quickMarcEditor.removeConfirmation.isPresent).to.be.true;
+    });
+  });
+
+  describe('remove record', () => {
```
Comment (NikitaSedyx) [inline test/bigtest/tests/QuickMarcEditor-test.js:77]: what about this?

Comment (uladislausamets) [reply test/bigtest/tests/QuickMarcEditor-test.js:77]: @NikitaSedyx  what is wrong in this case? We just  remove record in this test

Comment (NikitaSedyx) [reply test/bigtest/tests/QuickMarcEditor-test.js:77]: name, you updated previous one, why not to make case more readable?

**Pull request merged**